parser/varjson.go

```diff
@@ -8,23 +8,28 @@ import (
 )
 
 // Method VarJSON behaves in two ways depending
-// on whether `multipart` is true or not.
+// on whether `multipart` or `form` is true or not.
 // If there is no multipart, then VarJSON tries
 // to match one or more VarJSONPairs
-// However, if there is multipart, we try to match
+// However, if there is multipart or form, we try to match
 // zero or more VarJSON, followed by zero or more
 // file fields (separated by `@`). If there is no match
 // at all, we return a ParseError; otherwise the
 // we return the parsed data.
 func (p *Lama2Parser) VarJSON() (*gabs.Container, error) {
 	temp := gabs.New()
 	hasMultipart := false
+	hasForm := false
 	if val, ok := p.Context["multipart"]; ok {
 		hasMultipart = val
 	}
 
+	if val, ok := p.Context["form"]; ok {
+		hasForm = val
+	}
+
 	pair, e1 := p.Match([]string{"VarJSONPair"})
-	if e1 != nil && !hasMultipart {
+	if e1 != nil && (!hasMultipart || !hasForm) {
```

Comment:
Can simplify condition. What are we checking for here? 

```
!a || !b = a && b
```

Reply:
![image](https://github.com/HexmosTech/Lama2/assets/796107/bf375454-a2ea-4ece-b664-aae94691dd9a)


![image](https://github.com/HexmosTech/Lama2/assets/796107/9180cdd4-1a2e-4801-99a5-0522f188acd9)

If either a or b is false, then the result is true.

Reply:
Need a comment explaining the whole thing with simplification.

Reply:
When i rechecked that part, Since FORM does not have file upload, the condition is not needed, since that condition is for erroring out in the event where there is no key value pairs.

multipart has files so it can be executed even without key value pairs, so that's why that condition is there only for multipart